src/app/shared/classes/test-flow-base-pages/office/__tests__/office-base-page.spec.ts

```diff
@@ -76,4 +135,60 @@ describe('OfficeBasePageComponent', () => {
     });
   });
 
+  describe('setIsValidStartDateTime', () => {
+    it('should set isValidStartDateTime', () => {
+      basePageComponent.setIsValidStartDateTime(true);
+      expect(basePageComponent.isValidStartDateTime).toEqual(true);
+    });
+  });
+
+  describe('weatherConditionsChanged', () => {
+    it('should dispatch a weather conditions changed action with the weather condition values', () => {
+      const conditions: WeatherConditions[] = ['Showers'];
+      basePageComponent.weatherConditionsChanged(conditions);
+      expect(store$.dispatch).toHaveBeenCalledWith(WeatherConditionsChanged(conditions));
+    });
+  });
+
+  describe('activityCodeChanged', () => {
+    it('should dispatch a SetActivityCode action with the activity code', () => {
+      basePageComponent.activityCodeChanged(activityCodeModelList[0]);
+      expect(store$.dispatch).toHaveBeenCalledWith(SetActivityCode(activityCodeModelList[0].activityCode));
+    });
+  });
+
+  describe('completeTest', () => {
+    it('should successfully end the test', () => {
+      basePageComponent.completeTest();
+      expect(store$.dispatch).toHaveBeenCalledWith(CompleteTest());
+    });
+
+    it('should not dispatch complete test if in practice mode', () => {
+      basePageComponent.isPracticeMode = true;
+      basePageComponent.completeTest();
+      expect(store$.dispatch).not.toHaveBeenCalledWith(CompleteTest());
+    });
+  });
+
+  describe('popToRoot', () => {
+    it('should call the navigateBack method in the navcontroller if not in practice mode', () => {
+      basePageComponent.popToRoot();
+      expect(basePageComponent.navController.navigateBack).toHaveBeenCalled();
+    });
+    it('should call the navigateBack method in the navcontroller if in practice mode.', () => {
+      basePageComponent.isPracticeMode = true;
+      basePageComponent.popToRoot();
+      expect(basePageComponent.navController.navigateBack).toHaveBeenCalled();
+    });
+  });
+
+  describe('defer', () => {
+    it('should call popToRoot and dispatch two actions', async () => {
+      spyOn(basePageComponent, 'popToRoot');
+      await basePageComponent.defer();
+      expect(basePageComponent.popToRoot).toHaveBeenCalled();
+      expect(store$.dispatch).toHaveBeenCalledWith(SavingWriteUpForLater());
+      expect(store$.dispatch).toHaveBeenCalledWith(PersistTests());
+    });
+  });
 });
```

Comment:
There are a lot of untested methods in this file - might be worth leaving a TODO to add some coverage. I can see some of the tests in mes-mobile-app exist but within each spec so would be the case of just porting those over for some of these methods

Reply:
I didnt like leaving the file untested, but following our agreed pattern of filling in the tech debt at a later date.  Will add a TODO to that effect and story to match.

Reply:
Its not tech debt if the tests already exist surely?